example/lib/main.dart

```diff
@@ -1,11 +1,101 @@
 import 'package:flutter/material.dart';
 import 'package:easy_pdf_viewer/easy_pdf_viewer.dart';
 
-void main() => runApp(MyApp());
+void main() => runApp(MyApp(
+      progressExample: true,
+    ));
 
 class MyApp extends StatefulWidget {
+  const MyApp({this.progressExample = false});
+
+  final bool progressExample;
+
+  @override
+  State<MyApp> createState() {
+    if (progressExample) return _MyAppStateWithProgress();
+    return _MyAppState();
+  }
+}
+
+class _MyAppStateWithProgress extends State<MyApp> {
+  bool _isLoading = true;
+  PDFDocument document;
+  DownloadProgress downloadProgress;
+
+  @override
+  void initState() {
+    loadDocument();
+    super.initState();
+  }
+
   @override
-  _MyAppState createState() => _MyAppState();
+  void setState(VoidCallback fn) {
+    if (mounted) {
+      super.setState(fn);
+    }
+  }
+
+  void loadDocument() async {
+
+    /// Clears the cache before download, so [PDFDocument.fromURLWithDownloadProgress.downloadProgress()]
+    /// is always executed (meant only for testing).
+    await DefaultCacheManager().emptyCache();
+
+    PDFDocument.fromURLWithDownloadProgress(
+      'https://www.ecma-international.org/wp-content/uploads/ECMA-262_12th_edition_june_2021.pdf',
+      downloadProgress: (downloadProgress) => setState(() {
+        this.downloadProgress = downloadProgress;
+      }),
+      onDownloadComplete: (document) => setState(() {
+        this.document = document;
+        _isLoading = false;
+      }),
+    );
+  }
+
+  Widget buildProgress() {
+    if (downloadProgress == null) return SizedBox();
+
+    String parseBytesToKBs(int bytes) {
+      return '${(bytes / 1000).toStringAsFixed(2)} KBs';
+    }
+
+    String progressString =
```

Comment:
Double `KBs`, maybe we should remove one.

Reply:
Oh missed it! 😄